src/main/java/bio/terra/cli/context/WorkspaceContext.java

```diff
@@ -189,18 +186,85 @@ public void removeCloudResource(String name) {
   }
 
   // ====================================================
-  // Directory and file names
-  //   - current working directory: .
-  //   - persisted workspace context file: ./terra-cli/workspace_context.json
-  //   - sub-directories for tools (e.g. ./nextflow) are defined in the SupportedToolHelper
-  // sub-classes
+  // Resolving file and directory paths
+
+  /**
+   * Get the workspace directory. (i.e. the parent of the .terra directory)
+   *
+   * @return the absolute path to the workspace directory
+   */
+  @SuppressFBWarnings(
+      value = "NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE",
+      justification =
+          "An NPE would only happen here if there was an error getting the workspace context file, and an exception would be thrown in that method instead.")
+  @JsonIgnore
+  public static Path getWorkspaceDir() {
+    return getWorkspaceContextFile().getParent().getParent();
+  }
+
+  /**
+   * Get a handle for the workspace context file.
+   *
+   * <p>This method first searches for an existing workspace context file in the current directory
+   * hierarchy. If it finds one, then it returns a handle to that file.
+   *
+   * <p>If it does not find one, then it returns a handle to the file where a new workspace context
+   * can be written. This handle will be relative to the current directory (i.e. current directory =
+   * workspace top-level directory)
+   *
+   * @return a handle for the workspace context file
+   */
+  @JsonIgnore
+  private static Path getWorkspaceContextFile() {
+    Path currentDir = Path.of("").toAbsolutePath();
+    try {
+      return getExistingWorkspaceContextFile(currentDir);
+    } catch (FileNotFoundException fnfEx) {
+      return currentDir.resolve(WORKSPACE_CONTEXT_DIRNAME).resolve(WORKSPACE_CONTEXT_FILENAME);
+    }
+  }
 
   /**
-   * Getter for the file where the workspace context is persisted.
+   * Get a handle for an existing workspace context file in the current directory hierarchy.
```

Comment:
does this return an absolute path or a relative path?

Reply:
Absolute. Updated comment.